src/main/java/org/cyclonedx/util/ExtensionSerializer.java

```diff
@@ -0,0 +1,135 @@
+          staxWriter.writeStartElement(vulns.getPrefix(), "rating", vulns.getNamespaceURI());
+          if (r.getScore() != null) {
+            staxWriter.writeStartElement(vulns.getPrefix(), "score", vulns.getNamespaceURI());
+            generateTextNode(staxWriter, "base", r.getScore().getBase().toString(), vulns.getNamespaceURI(), vulns.getPrefix());
+            generateTextNode(staxWriter, "impact", r.getScore().getImpact().toString(), vulns.getNamespaceURI(), vulns.getPrefix());
+            generateTextNode(staxWriter, "exploitability", r.getScore().getExploitability().toString(), vulns.getNamespaceURI(), vulns.getPrefix());
+            staxWriter.writeEndElement();
+          }
+          generateTextNode(staxWriter, "severity", r.getSeverity().getSeverityName(), vulns.getNamespaceURI(), vulns.getPrefix());
+          generateTextNode(staxWriter, "method", r.getMethod().getScoreSourceName(), vulns.getNamespaceURI(), vulns.getPrefix());
+          generateTextNode(staxWriter, "vector", r.getVector(), vulns.getNamespaceURI(), vulns.getPrefix());
+          staxWriter.writeEndElement();
+        }
+        staxWriter.writeEndElement();
+      }
+      if (vuln.getCwes() != null && !vuln.getCwes().isEmpty()) {
+        staxWriter.writeStartElement(vulns.getPrefix(), "cwes", vulns.getNamespaceURI());
+        for (Cwe c : vuln.getCwes()) {
+          generateTextNode(staxWriter, "cwe", c.getText().toString(), vulns.getNamespaceURI(), vulns.getPrefix());
+        }
+        staxWriter.writeEndElement();
+      }
+      generateTextNode(staxWriter, "description", vuln.getDescription(), vulns.getNamespaceURI(), vulns.getPrefix());
+      if (vuln.getRecommendations() != null && !vuln.getRecommendations().isEmpty()) {
+        staxWriter.writeStartElement(vulns.getPrefix(), "recommendations", vulns.getNamespaceURI());
+        for (Recommendation r : vuln.getRecommendations()) {
+          generateTextNode(staxWriter, "recommendation", r.getText(), vulns.getNamespaceURI(), vulns.getPrefix());
+        }
+        staxWriter.writeEndElement();
+      }
+      if (vuln.getAdvisories() != null && !vuln.getAdvisories().isEmpty()) {
+        staxWriter.writeStartElement(vulns.getPrefix(), "advisories", vulns.getNamespaceURI());
+        for (Advisory a : vuln.getAdvisories()) {
+          generateTextNode(staxWriter, "advisory", a.getText(), vulns.getNamespaceURI(), vulns.getPrefix());
+        }
+        staxWriter.writeEndElement();
+      }
+      staxWriter.writeEndElement();
+    }
+    staxWriter.writeEndElement();
+  }
+
+
+  private void generateTextNode(final XMLStreamWriter writer,
+                                final String fieldName,
+                                final String value,
+                                final String namespaceUri,
+                                final String prefix) throws XMLStreamException
+  {
+    writer.writeStartElement(prefix, fieldName, namespaceUri);
```

Comment:
Is the namespace needed for all the fields? should we just add it to the main one "vulnerabilities"?

Reply:
IIRC the namespace is needed on all elements otherwise I THINK Jackson generates it's own wackadoo one.